Library/Homebrew/dev-cmd/extract.rb

```diff
@@ -0,0 +1,153 @@
+#:  * `extract` [`--force`] <formula> `--tap=`<tap> [`--version=`<version>]:
+#:    Looks through repository history to find the <version> of <formula> and
+#:    creates a copy in <tap>/Formula/<formula>@<version>.rb. If the tap is
+#:    not installed yet, attempts to install/clone the tap before continuing.
+#:    A tap must be passed through `--tap` in order for `extract` to work.
+#:
+#:    If `--force` is passed, the file at the destination will be overwritten
+#:    if it already exists. Otherwise, existing files will be preserved.
+#:
+#:    If an argument is passed through `--version`, <version> of <formula>
+#:    will be extracted and placed in the destination tap. Otherwise, the most
+#:    recent version that can be found will be used.
+
+require "utils/git"
+require "formulary"
+require "tap"
+
+class BottleSpecification
+  def method_missing(m, *_args, &_block)
+    if [:sha1, :md5].include?(m)
+      opoo "Formula has unknown or deprecated stanza: #{m}" if ARGV.debug?
+    else
+      super
+    end
+  end
+end
+
+class Module
```

Comment:
Dont' monkey-patch this.

Reply:
@reitermarkus Do you have an alternate suggestion to handling old formulae?

Reply:
We can use refinements to limit the monkey-patch to a certain scope.

Reply:
Seems like a good idea 👍